qiskit/transpiler/passes/optimization/split_2q_unitaries.py

```diff
@@ -24,17 +25,39 @@ class Split2QUnitaries(TransformationPass):
     matrix is actually a product of 2 single qubit gates. In these cases the 2q gate can be
     simplified into two single qubit gates and this pass will perform this optimization and will
     replace the two qubit gate with two single qubit :class:`.UnitaryGate`.
+
+    If some of the gates can be viewed as a swap joined by the product of 2 single qubit gates,
+    the pass will recreate the DAG, permuting the swapped qubits similar
+    to how it's done in :class:`ElidePermutations`.
     """
 
-    def __init__(self, fidelity: float = 1.0 - 1e-16):
+    def __init__(self, fidelity: float = 1.0 - 1e-16, split_swap: bool = False):
```

Comment:
By default this means we don't run the swap splitting, are you planning a follow-up where you update https://github.com/Qiskit/qiskit/blob/main/qiskit/transpiler/preset_passmanagers/builtin_plugins.py#L192-L194 to set it to True unless `routing=None` is specified?

Reply:
That's exactly what I'm wondering about. If you think the specific logic you suggested here suffices, I'll add this as part of this PR.

Reply:
Yeah I think that's sufficient. You can mirror the exact logic that conditions the execution of `ElidePermutations` .

Reply:
Done in a9bc3e1ea